aspnetcore/fundamentals/logging/index/samples/3.x/TodoApiSample/Program.cs

```diff
@@ -0,0 +1,193 @@
+using TodoApiSample.Infrastructure;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging.AzureAppServices;
+using Microsoft.Extensions.Logging.Debug;
+using Microsoft.Extensions.Hosting;
+
+namespace TodoApiSample
+{
+    public class Program
+    {
+#if TemplateCode
+        #region snippet_TemplateCode
+        public static void Main(string[] args)
+        {
+            CreateHostBuilder(args).Build().Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseStartup<Startup>();
+                });
+        #endregion
+#elif LogFromMain
+        #region snippet_LogFromMain
+        public static void Main(string[] args)
+        {
+            var host = CreateHostBuilder(args).Build();
+
+            var todoRepository = host.Services.GetRequiredService<ITodoRepository>();
+            todoRepository.Add(new Core.Model.TodoItem() { Name = "Feed the dog" });
+            todoRepository.Add(new Core.Model.TodoItem() { Name = "Walk the dog" });
+
+            var logger = host.Services.GetRequiredService<ILogger<Program>>();
+            logger.LogInformation("Seeded the database.");
+
+            IMyService myService = host.Services.GetRequiredService<IMyService>();
+            myService.WriteLog("Logged from MyService.");
+
+            host.Run();
+        }
+        #endregion
+
+        #region snippet_AddProvider
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureLogging(logging =>
+                {
+                    logging.ClearProviders();
+                    logging.AddConsole();
+                })
+                .ConfigureWebHostDefaults(webBuilder =>
+                {
+                    webBuilder.UseStartup<Startup>();
+                });
+        #endregion
+#elif AzLogOptions
+        #region snippet_AzLogOptions
+        public static void Main(string[] args)
+        {
+            var host = CreateHostBuilder(args).Build();
+
+            var todoRepository = host.Services.GetRequiredService<ITodoRepository>();
+            todoRepository.Add(new Core.Model.TodoItem() { Name = "Feed the dog" });
+            todoRepository.Add(new Core.Model.TodoItem() { Name = "Walk the dog" });
+
+            var logger = host.Services.GetRequiredService<ILogger<Program>>();
+            logger.LogInformation("Seeded the database.");
+
+            host.Run();
+        }
+
+        public static IHostBuilder CreateHostBuilder(string[] args) =>
+            Host.CreateDefaultBuilder(args)
+                .ConfigureLogging(logging => logging.AddAzureWebAppDiagnostics())
+                    .ConfigureServices(serviceCollection => serviceCollection
+                    .Configure<AzureFileLoggerOptions>(options =>
```

Comment:
The formatting/indentation is off here. .Configure should be intended one more level and the closing paren should be on its own line.